test/tests/pbs_accumulate_resc_used.py

```diff
@@ -0,0 +1,1456 @@
+        s = self.server.accounting_match(
+            "E;%s;.*%s.*" % (jid, acctlog_match), regexp=True, n=100)
+        self.assertTrue(s)
+
+        acctlog_match = 'resources_used.ncpus=1'
+        s = self.server.accounting_match(
+            "E;%s;.*%s.*" % (jid, acctlog_match), regexp=True, n=100)
+        self.assertTrue(s)
+
+        acctlog_match = 'resources_used.foo_str3='
+        acctlog_match += '\{"a"\:6\,"b"\:"some value \#\$\%\^\&\*\@"\,'
+        acctlog_match += '"c"\:54\.4\,"d"\:"32\.5gb"\}'
+        s = self.server.accounting_match(
+            "E;%s;.*%s.*" % (jid, acctlog_match), regexp=True, n=100)
+        self.assertTrue(s)
+        acctlog_match = 'resources_used.stra=\"glad\,elated\"\,\"happy\"'
+        s = self.server.accounting_match(
+            "E;%s;.*%s.*" % (jid, acctlog_match), regexp=True, n=100)
+        self.assertTrue(s)
+
+    def test_prologue(self):
+        """
+        Test accumulatinon of resources of a multinode job from an
+        exechost_prologue hook.
+        """
+        self.logger.info("test_prologue")
+        hook_body = """
+import pbs
+e=pbs.event()
+pbs.logmsg(pbs.LOG_DEBUG, "executed prologue hook")
+if e.job.in_ms_mom():
+    e.job.resources_used["vmem"] = pbs.size("11gb")
+    e.job.resources_used["foo_i"] = 11
+    e.job.resources_used["foo_f"] = 0.11
+    e.job.resources_used["foo_str"] = '{"seven":7}'
+    e.job.resources_used["cput"] = 11
+    e.job.resources_used["stra"] = '"glad,elated","happy"'
+    e.job.resources_used["foo_str3"] = \
+      \"\"\"{"a":6,"b":"some value #$%^&*@","c":54.4,"d":"32.5gb"}\"\"\"
+    e.job.resources_used["foo_str2"] = "seven"
+    e.job.resources_used["foo_str4"] = "eight"
+else:
+    e.job.resources_used["vmem"] = pbs.size("12gb")
+    e.job.resources_used["foo_i"] = 12
+    e.job.resources_used["foo_f"] = 0.12
+    e.job.resources_used["foo_str"] = '{"eight":8,"nine":9}'
+    e.job.resources_used["foo_str2"] = '{"seven":7}'
+    e.job.resources_used["cput"] = 12
+    e.job.resources_used["stra"] = '"cucumbers,bananas"'
+    e.job.resources_used["foo_str3"] = \"\"\""vn1":4,"vn2":5,"vn3":6\"\"\"
+"""
+
+        hook_name = "prolo"
+        a = {'event': "execjob_prologue", 'enabled': 'True'}
+        rv = self.server.create_import_hook(
+            hook_name,
+            a,
+            hook_body,
+            overwrite=True)
+        self.assertTrue(rv)
+
+        a = {'job_history_enable': 'True'}
+        self.server.manager(MGR_CMD_SET, SERVER, a, expect=True)
+
+        a = {'Resource_List.select': '3:ncpus=1', 'Resource_List.walltime': 10}
+        j = Job(TEST_USER)
+        j.set_attributes(a)
+        j.set_sleep_time("10")
+
+        # The pbsdsh call is what allows a first task to get spawned on
+        # on a sister mom, causing the execjob_prologue hook to execute.
+        j.create_script(
+            "pbsdsh -n 1 hostname\n" + "pbsdsh -n 2 hostname\n" + "sleep 10\n")
```

Comment:
If you create a job script, do you need to set the sleep time?  Doesn't that just set the sleep time for a default job script?